action.yml

```
@@ -162,8 +162,13 @@ runs:
           FILES=$(jq -r .files[].filename __compare.json)
           rm __compare.json
 
-          # Don't allow merge-ups when there are changes in javascript dependency files
-          DEPENDENCY_FILES="package.json yarn.lock"
+          # Don't allow merge-ups when there are changes in javascript dependency files (e.g. package.json)
+          # Note that yarn.lock is allowed to merge-up, including between majors, because
+          # its presence in a git diff will force a yarn build which will essentially delete
```

```suggestion
          # its presence in a git diff will force a `yarn install` and `yarn build` which will essentially delete
```
It's the install that really matters here I think.

Also make sure `yarn install` is actually run (looking at the code below it looks like it might not be? So that relies on the `yarn build` script in package.json starting with `yarn &&` which maybe they all do right now but we shouldn't rely on that for this.

---

Ah yup, good point. I made some bad assumptions here. Also overwritting yarn.lock would result in a essentially doing new JS deps upgrade on merge-up, which isn't what we want.

Have updated to disallow merge-up on major